src/main/java/filter/expression/Qualifier.java

```diff
@@ -117,6 +148,36 @@ public static boolean process(IModel model, FilterExpression expr, TurboIssue is
         return exprWithNormalQualifiers.isSatisfiedBy(model, issue, new MetaQualifierInfo(metaQualifiers));
     }
 
+    private static Optional<Integer> getCurrentMilestoneIndex(List<TurboMilestone> allMilestones) {
+        if (allMilestones.isEmpty()) {
+            return Optional.empty();
+        }
+
+        int currentIndex = 0;
+
+        for (TurboMilestone checker : allMilestones) {
+            // current milestone must be open
+            if (checker.isOpen()) {
+                // if milestone is overdue but have no open issue, ignore
+                if (checker.getDueDate().isPresent() &&
+                        checker.getDueDate().get().isBefore(LocalDate.now()) &&
+                        checker.getOpenIssues() == 0) {
+                    // do nothing; using negative statement here
+                    // since positive statement would be much longer
+                    // and more complicated; this works since
+                    // milestones are sorted
```

Comment:
How about this?
```java
boolean overdue = checker.getDueDate().isPresent()
    && checker.getDueDate().get().isBefore(LocalDate.now());

boolean relevant = !(overdue && checker.getOpenIssues() > 0);

if (checker.isOpen() && relevant) {
    return Optional.of(currentIndex);
}
```
This isn't super important, just trying to write it so the comments aren't needed.